crates/ruff/src/rules/flake8_annotations/rules.rs

```diff
@@ -150,6 +373,25 @@ impl Violation for MissingReturnTypeClassMethod {
 }
 
 define_violation!(
+    /// ### What it does
+    /// Checks that an expression is annotated with a more specific type than `Any`.
+    ///
+    /// ### Why is this bad?
+    /// `Any` is a type that can be anything, and it is the default type for
+    /// unannotated expressions. It is better to be explicit about the type of an
+    /// expression, and to use `Any` only when it is really needed.
```

Comment:
>[Any] is the default type for unannotated expressions.

While that's true for mypy (and probably also pyright?) it's not generally true, e.g. Pyright uses [Unknown](https://github.com/microsoft/pyright/blob/main/docs/internals.md#type-checking-concepts) instead.

Reply:
Wow TIL

Reply:
(Did you mean "probably also pyre"? Or "pytype"? I see pyright twice in that sentence.)

Reply:
(ah yes I meant pytype ...)

Reply:
I should add that Pyright has very good type inference so for most expressions it can actually infer the right type without any annotations ... but when a function return type cannot be inferred, it infers `Unknown` instead of `Any`.